runtime/doc/lua.txt

```diff
@@ -34,20 +34,32 @@ Nvim ever ships with Lua 5.4+, a Lua 5.1 compatibility shim will be provided
 so that old plugins continue to work transparently.
 
                                                                   *lua-luajit*
-Nvim is built with luajit on platforms which support it, which provides
-extra functionality. Lua code in |init.lua| and plugins can assume its presence
-on installations on common platforms. For maximum compatibility with less
-common platforms, availability can be checked using the `jit` global variable: >lua
+On supported platforms, Nvim is built with LuaJIT, which provides extra
+functionality (compared to PUC Lua) such as "bit" and various utilities in
+`$VIMRUNTIME/lua/jit/`. Lua code in |init.lua| and plugins can assume its
+presence on many platforms, but for maximum compatibility should check the
+`jit` global variable: >lua
   if jit then
     -- code for luajit
   else
     -- code for plain lua 5.1
   end
 <
                                                                   *lua-bit*
-In particular, the luajit "bit" extension module is _always_ available.
-A fallback implementation is included when nvim is built with PUC Lua 5.1,
-and will be transparently used when `require("bit")` is invoked.
+The LuaJIT "bit" extension module is _always_ available: when built with PUC
+Lua, Nvim includes a fallback implementation which provides `require("bit")`.
+
+                                                                  *lua-profile*
+To profile Lua code (with LuaJIT-enabled Nvim), the basic steps are: >lua
+    -- Start a profiling session:
+    require('jit.p').start('i1', '/tmp/profile')
```

Comment:
We should consider vendoring the latter (with neovim-specific tweaks); profiling Lua code is such an important dev tool that the lack of it in the standard library is hurting. 

Reply:
The code between start-stop ran too fast for a sample-based profiler, so no samples were collected. Running these lines repetitively (e.g. `for i = 1,10000 do`) would give some results.

Reply:
Should we add a real profiler? Either import from existing implementation or rewrite from scratch with this at the backend? This feels like it could be streamlined.

Reply:
Yes we should (but not now)

Reply:
> Should we add a real profiler?

What properties does a real profiler have that `jit.p` lacks?

Reply:
Actual usability.

Reply:
It was an offhand comment about the the clunkiness of jit.p, not a RFC for the government. I am not ready to defend this that seriously yet.

Reply:
But for starters the flags are finicky to get right. At least some wrapper around those with "sane" defaults would be nice.

Reply:
To be clear, every profiler uses jit.p under the hood (and hence doesn't work on PUC Lua); the question is just end-user usability (which jit.p does _not_ have).

But this is not the place for this discussion; the point is that recommending jit.p in documentation right now is a bad idea, and we should postpone the whole discussion to a separate issue after 0.10.

Reply:
> But for starters the flags are finicky to get right. At least some wrapper around those with "sane" defaults would be nice.

I'm not sure there are saner defaults. With profilers you usually need to have some idea what you're looking for. It also helps to keep in mind that all profilers are liers. It's often a good idea to gather different kind of data.
It wouldn't surprise me if the `debug.sethook(` based variant yields different results than the luajit profiler